vectorstores/chromadb/chroma.go

```diff
@@ -0,0 +1,187 @@
+package chromadb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	chroma "github.com/amikos-tech/chroma-go"
+	chromaopenapi "github.com/amikos-tech/chroma-go/swagger"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+var (
+	// ErrEmbedderWrongNumberVectors is returned when if the embedder returns a number
+	// of vectors that is not equal to the number of documents given.
+	ErrEmbedderWrongNumberVectors = errors.New(
+		"number of vectors from embedder does not match number of documents",
+	)
+	ErrInvalidScoreThreshold = errors.New(
+		"score threshold must be between 0 and 1")
+)
+
+// Store is a wrapper around the chromadb client.
+type Store struct {
+	embedder wrappedEmbeddingFunction
+	client   *chroma.Client
+
+	scheme string
+	host   string
+
+	// optional
+	nameSpaceKey string
+	// optional
+	textKey string
+	// optional: nameSpace represents collection in chromadb
+	nameSpace string
+	// optional
+	distanceFunc chroma.DistanceFunction
+	// optional
+	includes []chroma.QueryEnum
+}
+
+var _ vectorstores.VectorStore = Store{}
+
+// New creates a new Store with options for chromadb.
+func New(opts ...Option) (vectorstores.VectorStore, error) {
+	s, err := applyClientOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := chromaopenapi.NewConfiguration()
+	configuration.Servers = chromaopenapi.ServerConfigurations{
+		{
+			URL:         fmt.Sprintf("%s://%s", s.scheme, s.host),
+			Description: "Chromadb server url for this store",
+		},
+	}
+	s.client = &chroma.Client{
+		ApiClient: chromaopenapi.NewAPIClient(configuration),
+	}
+
+	if _, err = s.client.Heartbeat(); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (s Store) AddDocuments(ctx context.Context, docs []schema.Document, options ...vectorstores.Option) error {
+	opts := s.getOptions(options...)
+	nameSpace := s.getNameSpace(opts)
+
+	texts := make([]string, 0, len(docs))
+	ids := make([]string, len(docs))
+	for idx, doc := range docs {
+		texts = append(texts, doc.PageContent)
+		ids[idx] = fmt.Sprintf("%d", idx)
+	}
+
+	collection, err := s.client.CreateCollection(s.nameSpace, map[string]interface{}{}, true, s.embedder, s.distanceFunc)
```

Comment:
I struggled with the idea of using the adapter's `NameSpace` concept to map directly to Chroma "Collections."  I resisted the temptation initially, almost gave in, then realized I believe even more strongly that they're separate things and that I believe we shouldn't "hide" Chroma collections behind the concept of `NameSpaces` in this API.

Please feel free to disagree, but consider my rationalizations:

1. Weaviate doesn't have "Namespaces" yet the idea is implemented using its other attributes for "filtering" objects (though I don't understand these well) such as "Class" and "Module", etc.   `NameSpace` seems to be implemented in that adapter using metadata filters, etc. which I felt could be done also in the Chroma adapter.
2. I've seen cases where Pinecone "NameSpaces" are composed - that is, a set of objects is defined to comprise a superset of _several_ namespaces.  That'd be hard to do in Chroma - e.g., search across several collections?

In any case, I'm very interested in hearing your feedback on this subject.

Reply:
@noodnik2 seeing your comments to make me feel that we  do have struggled with the same things.... 
 
I changed many times on this `collection` or `namespace`... better to  request some comments from the langchain matainer.